src/main/sensors/gyro_init.c

```diff
@@ -603,66 +607,88 @@ bool gyroInit(void)
         eepromWriteRequired = true;
     }
 
-#if defined(USE_MULTI_GYRO)
-    if ((gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH && !((gyroDetectionFlags & GYRO_ALL_MASK) == GYRO_ALL_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_1 && !(gyroDetectionFlags & GYRO_1_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_2 && !(gyroDetectionFlags & GYRO_2_MASK))) {
-        if (gyroDetectionFlags & GYRO_1_MASK) {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_1;
-        } else {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_2;
+#if GYRO_COUNT > 1
+    uint8_t gyro_all_mask = GYRO_NONE_MASK;
+
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        gyro_all_mask |= GYRO_MASK(i);
+    }
+
+    bool not_all_gyro = false;
+    for (int i = 0; i < GYRO_COUNT; i++) {
+        if (gyro.gyroToUse == (i) && !(gyroDetectionFlags & GYRO_MASK(i))) {
+            not_all_gyro = true;
+        }
+    }
+
+    if ((gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_ALL && !((gyroDetectionFlags & gyro_all_mask) == gyro_all_mask))
```

Comment:
You probably want test `popcount(gyroDetectionFlags) == 1` (only 1 gyro detected). Otherwise, mask should be used instead of gyroToUse in most places.
And you can always use mask, removing a lot of code duplication